src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1019,14 +1020,22 @@ protected void ExecuteIndexing()
                                     if (ShouldReplace())
                                     {
                                         var originalName = Name.Replace(Constants.Documents.Indexing.SideBySideIndexNamePrefix, string.Empty, StringComparison.InvariantCultureIgnoreCase);
+                                        _isReplacing = true;
 
-                                        // this can fail if the indexes lock is currently held, so we'll retry
-                                        // however, we might be requested to shutdown, so we want to skip replacing
-                                        // in this case, worst case scenario we'll handle this in the next batch
-                                        while (_indexingProcessCancellationTokenSource.IsCancellationRequested == false)
+                                        try
+                                        {
+                                            // this can fail if the indexes lock is currently held, so we'll retry
+                                            // however, we might be requested to shutdown, so we want to skip replacing
+                                            // in this case, worst case scenario we'll handle this in the next batch
+                                            while (_indexingProcessCancellationTokenSource.IsCancellationRequested == false)
```

Comment:
Can this ever be an infinite loop?

Reply:
I'm pretty sure that it can't.

Reply:
As an aside, I looked at `TryReplaceIndexes`, it _never_ returns false?!